music21/romanText/writeRoman.py

```diff
@@ -0,0 +1,408 @@
+    >>> testInt
+    1
+
+    >>> testFrac = romanText.writeRoman.intBeat(8 / 3, roundValue=2)
+    >>> testFrac
+    2.67
+
+    >>> testStr = romanText.writeRoman.intBeat('0.666666666', roundValue=2)
+    >>> testStr
+    0.67
+
+    The roundValue sets the number of decimal places to round to. The default is two:
+
+    >>> testRound2 = romanText.writeRoman.intBeat(1.11111111, roundValue=2)
+    >>> testRound2
+    1.11
+
+    But this can be set to any integer:
+
+    >>> testRound1 = romanText.writeRoman.intBeat(1.11111111, roundValue=1)
+    >>> testRound1
+    1.1
+
+    '''
+
+    options = (str, int, float, fractions.Fraction)
+
+    if not isinstance(beat, options):
+        raise TypeError(f'Beat, (currently {beat}) must be one of {options}.')
+
+    if isinstance(beat, int):
+        return beat
+
+    if type(beat) in [str, fractions.Fraction]:
+        beat = float(beat)
+
+    # Now beat is definitely a float
+    if int(beat) == beat:
+        return int(beat)
+    else:
+        return round(beat, roundValue)
+
+
+# ------------------------------------------------------------------------------
+
+class Test(unittest.TestCase):
+    '''
+    Tests for two analysis cases (the smallest rntxt files in the music21 corpus)
+    along with two test by modifying those scores.
+
+    Additional tests for the stand alone functions rnString and intBeat.
+    '''
+
+    def testTwoCorpusPiecesAndTwoCorruptions(self):
+
+        from music21 import corpus
+        from music21 import meter
+
+        scoreBach = corpus.parse('bach/choraleAnalyses/riemenschneider004.rntxt')  # Smallest file
+
+        rnaBach = RnWriter(scoreBach)
+        self.assertEqual(rnaBach.combinedList[-1], 'm10 b1 V6/V b2 V b3 I')  # NB b1
+
+        # --------------------
+
+        scoreBach.parts[0].measure(3).insert(0, meter.TimeSignature('10/8'))
+        scoreBach.parts[0].measure(3).insert(1, meter.TimeSignature('5/8'))
+
+        wonkyBach = RnWriter(scoreBach)
+
+        self.assertEqual(wonkyBach.combinedList[9], 'Time Signature: 10/8')
+        self.assertEqual(wonkyBach.combinedList[10],
+                         'Note: further time signature change(s) unprocessed: [\'5/8\']')
+
+        # --------------------
+
+        scoreMonte = corpus.parse('monteverdi/madrigal.3.8.rntxt')  # Smallest file
+        rnMonte = RnWriter(scoreMonte)
+
+        self.assertEqual(rnMonte.composer, 'Monteverdi')
+        self.assertEqual(rnMonte.title, 'La piaga c\'ho nel core')
+        self.assertEqual(rnMonte.combinedList[-1], 'm57 b1 I')  # NB b1
+
+        # --------------------
+
+        # (Re-)assign metadata in the normal way
+        scoreMonte.metadata.title = 'Fake title'
+        scoreMonte.metadata.movementNumber = 123456789
+        scoreMonte.metadata.movementName = 'Fake movementName'
+
+        adjustedMonte = RnWriter(scoreMonte)
+        self.assertEqual(adjustedMonte.title, 'Fake title - No.123456789: Fake movementName')
+
+    @unittest.expectedFailure
```

Comment:
`unittest.expectedFailure` is not quite right for testing expected behavior. It's more for when the code is broken and hasn't been fixed yet, or there is some dependency causing trouble. Here, you're testing expected behavior!

Better to use:
```
with self.assertRaises(TypeError):
    doStuffThatWillRaise()
```